### ISSUE TITLE
attachments multiple issues

### ISSUE THREAD

**Issue by ktsaou** [laurent22/joplin#48]
### Operating system

- [ ] Windows
- [ ] macOS
- [x] Linux
- [ ] Android
- [ ] iOS

### Application

- [x] Desktop
- [ ] Mobile
- [ ] Terminal

*If it is a bug if possible please provide a screenshot showing the problem*

*If there is an error, please copy and paste the full error message*

---

Hi,

thank you for joplin! nice work!

Issues related to attachments in Linux desktop client (installed with appimage in fedora 27):

1. The dialog to select a file opens **behind** the main window. Actually the dialog is probably not defined as a dialog, so it is not centered on top of joplin main window (the window manager attempts to find some free space on screen to position it, making the whole operation a little bit weird).

2. A strange notification appears when the dialog to select a file opens.

   ![image](https://user-images.githubusercontent.com/2662304/33483137-04f1d060-d6a5-11e7-806d-04d092bb30d6.png)

3. A file is selected, but nothing happens (nothing is attached).

4. The dialog does not support selecting multiple files, so to attach 10 files, you have to repeat the operation 10 times.

5. The directory the last file was selected from, is not remembered, so if you need to attach 10 files, you have to navigate to the directory 10 times (this may be related to  point 3 above - attachments do not work, so nothing is attached, and nothing is remembered).

6. **drag and drop** of files in joplin window is not supported. Generally drag and drop should be preferred when working with attachments.

**Comment by ktsaou** [comment]
Here is a video of the operation. I did it twice:

1. double-click the selected file
2. select the file, click `Open`

The following issues are visible:

i. a notification opens
ii. the dialog opens behind main window
iii. nothing is attached

![joplin-attachments](https://user-images.githubusercontent.com/2662304/33483688-b4907bdc-d6a7-11e7-9fe5-1f439ee51500.gif)

**Comment by laurent22** [comment]
Some of these issues might be system-specific, in particular the dialog behind the window and the strange notification. It's not happening on Windows, macOS or Ubuntu, which I guess are the officially supported OSes. However if you find out some way to fix this, let me know and I'll see if it can be implemented.

I've added support for multi-selection and I've fixed the bug you were having where nothing happen after you select a file. Remembering the last directory is normally built-in too but maybe not for Fedora, so I'm now explicitly specifying the default directory. For drag and drop, I'll check later if it can be done.
